py3dtilers/Common/tileset_creation.py

```diff
@@ -67,20 +67,18 @@ def __transform_node(node: 'GeometryNode', user_args, tree_centroid=np.array([0,
             for feature_list in node.get_features():
                 feature_list.scale_features(user_args.scale, tree_centroid)
 
-        centroid = node.feature_list.get_centroid()
-        offset = np.array([0, 0, 0]) if user_args.offset[0] == 'centroid' else centroid + np.array(user_args.offset)
+        offset = np.array([0, 0, 0]) if user_args.offset[0] == 'centroid' else np.array(user_args.offset)
 
         if not user_args.crs_in == user_args.crs_out:
             transformer = Transformer.from_crs(user_args.crs_in, user_args.crs_out)
-            tree_centroid = np.array(transformer.transform(tree_centroid[0], tree_centroid[1], tree_centroid[2]))
-            offset = np.array(transformer.transform(offset[0], offset[1], offset[2]))
+            tree_centroid = np.array(transformer.transform((tree_centroid + offset)[0], (tree_centroid + offset)[1], (tree_centroid + offset)[2]))
             for feature_list in node.get_features():
-                feature_list.change_crs(transformer)
+                feature_list.change_crs(transformer, offset)
 
         distance = node.feature_list.get_centroid() - tree_centroid
 
         for feature_list in node.get_features():
-            feature_list.translate_features(-feature_list.get_centroid())
+            feature_list.translate_features(-offset)
```

Comment:
Currently, we subtract the centroid from the features to obtain coordinates around 0,0,0 in the tile content, then we place the tile at the correct position with the `transform` attribute of the tile in the `tileset.json` (the tile content position is relative to the `transform` attribute). This "correct position" is the `offset = centroid + np.array(user_args.offset)` that we compute at line [L71](https://github.com/VCityTeam/py3dtilers/pull/162/files#diff-6a9e310d59656b80aed595e45ca15df44186c8c93f5ef21426f5bb19f0477174L71) and return at line [L87](https://github.com/VCityTeam/py3dtilers/pull/162/files#diff-6a9e310d59656b80aed595e45ca15df44186c8c93f5ef21426f5bb19f0477174R87)

Subtracting only the offset from coordinates, then setting the `transform` to this offset (with the `offset` returned at [L87](https://github.com/VCityTeam/py3dtilers/pull/162/files#diff-6a9e310d59656b80aed595e45ca15df44186c8c93f5ef21426f5bb19f0477174R87)) works but seems a bit confusing to me.

If the `transform` attribute is causing an issue with reprojection+translation, or with some projections, maybe we could just make the `transform` attribute optional and set tile contents to absolute positions (as proposed in #133). Doing so may cause precision issues (since glTF encodes coordinates as Float32). I'll do some tests and try to find the best solution. Let me know what you think about this

Reply:
Implemented review comments. With these changes reprojection works correctly.